File: tests/fixtures/patterns/php/negative.php
<?php
// Negative fixture: none of these should trigger security patterns.

function safe_query($pdo, $user) {
    $stmt = $pdo->prepare("SELECT * FROM users WHERE name = ?");
    $stmt->execute([$user]);
}

function safe_hash($data) {
    return hash("sha256", $data);
}

function safe_random() {
    return random_int(1, 100);
}

function safe_include() {
    include "config.php";
}

function safe_string_ops() {
    $x = "hello";
    $y = strtoupper($x);
    $z = strlen($y);
}

File: tests/fixtures/patterns/python/negative.py
# Negative fixture: none of these should trigger security patterns.

import subprocess
import hashlib

def safe_subprocess():
    # No shell=True
    subprocess.run(["ls", "-la"])

def safe_hash():
    hashlib.sha256(b"data")

def safe_literal_query(cursor):
    cursor.execute("SELECT COUNT(*) FROM users")

def safe_yaml_load(data):
    import yaml
    yaml.safe_load(data)

def safe_string_ops():
    x = "hello"
    y = x.upper()
    z = len(y)
